diagnosisApplication/src/main/interface_adapter/diagnosis/DiagnosisState.java

```diff
@@ -29,30 +33,109 @@ public DiagnosisState() {
+        this.diagnosis2Name = (String) diagnosis2.get("name");
+        this.diagnosis2Accuracy = (Integer) diagnosis2.get("accuracy");
+        this.diagnosis2Icd = (String) diagnosis2.get("icd");
+        this.diagnosis2ProfName = (String) diagnosis2.get("profName");
+        this.diagnosis2Specializations = (List<String>) diagnosis2.get("specializations");
+        numDiagnoses += 1;
     }
 
-    public void setDiagnosis3(HashMap<Integer, HashMap<String, Object>> diagnosis3) {
+    public void setDiagnosis3(HashMap<String, Object> diagnosis3) {
         this.diagnosis3Ranking = 3;
-        this.diagnosis3Name = diagnosis3.get("name");
-        this.diagnosis3Accuracy = diagnosis3.get("accuracy");
-        this.diagnosis3Icd = diagnosis3.get("icd");
-        this.diagnosis3ProfName = diagnosis3.get("profName");
-        this.diagnosis3Specializations = diagnosis3.get("specializations");
+        this.diagnosis3Name = (String) diagnosis3.get("name");
+        this.diagnosis3Accuracy = (Integer) diagnosis3.get("accuracy");
+        this.diagnosis3Icd = (String) diagnosis3.get("icd");
+        this.diagnosis3ProfName = (String) diagnosis3.get("profName");
+        this.diagnosis3Specializations = (List<String>) diagnosis3.get("specializations");
+        numDiagnoses += 1;
+    }
+
+    public int getNumDiagnoses() {
+        return numDiagnoses;
+    }
+
+    public Integer getDiagnosis1Ranking() {
+        return diagnosis1Ranking;
+    }
+
+    public String getDiagnosis1Name() {
+        return diagnosis1Name;
+    }
+
+    public Integer getDiagnosis1Accuracy() {
+        return diagnosis1Accuracy;
+    }
+
+    public String getDiagnosis1Icd() {
+        return diagnosis1Icd;
+    }
+
+    public String getDiagnosis1ProfName() {
+        return diagnosis1ProfName;
+    }
+
+    public List<String> getDiagnosis1Specializations() {
+        return diagnosis1Specializations;
+    }
+
+    public Integer getDiagnosis2Ranking() {
+        return diagnosis1Ranking;
+    }
+
+    public String getDiagnosis2Name() {
+        return diagnosis1Name;
+    }
+
+    public Integer getDiagnosis2Accuracy() {
+        return diagnosis1Accuracy;
+    }
+
+    public String getDiagnosis2Icd() {
+        return diagnosis1Icd;
+    }
+
+    public String getDiagnosis2ProfName() {
+        return diagnosis1ProfName;
+    }
+
+    public List<String> getDiagnosis2Specializations() {
+        return diagnosis1Specializations;
+    }
+
+    public Integer getDiagnosis3Ranking() {
+        return diagnosis1Ranking;
+    }
+
+    public String getDiagnosis3Name() {
+        return diagnosis1Name;
+    }
+
+    public Integer getDiagnosis3Accuracy() {
+        return diagnosis1Accuracy;
+    }
+
+    public String getDiagnosis3Icd() {
+        return diagnosis1Icd;
+    }
+
+    public String getDiagnosis3ProfName() {
+        return diagnosis1ProfName;
+    }
+
+    public List<String> getDiagnosis3Specializations() {
+        return diagnosis1Specializations;
```

Comment:
Setting up getters to pull the diagnosis output from the state so the view doesn't have information stored in order to follow clean architecture.